### ISSUE TITLE
Sony Xperia T, From 2.8.7.0 to latest version

### ISSUE THREAD

**Issue by jakubpolo** [TeamWin/Team-Win-Recovery-Project#997]
Hello,
I am experiencing problems with updating my TWRP from 2.8.7.0 to latest version. 

I am using custom ROM and have root access. 

I downloaded the app, downloaded the image for my phone Sony Xperia T (twrp-3.1.1-0-mint.img), selected it and flashed to recovery. After that it says "Flash Completed Successfully!" but when I restart to TWRP I can see that I still have 2.8.7.0 version.

The same happens when I install the image from within the TWPR. Again everything seems to be OK, full of green sings but the version of my TWRP is still 2.8.7.0.

I need to update it to try no ROM and I was unable to google my self to answers what might be the problem.

Thank you for your time.

**Comment by nailyk-fr** [comment]
It is a rom issue. 
Sony had great bootloader who didnt allow direct boot to recovery. 
So the kernel is in charge of this. On stock rom you cannot change the kernel because it is signed. 
Ensure your ROM support extract_elf_ramdisk, unless you will never be able to boot FOTAKernel recovery.

**Comment by jakubpolo** [comment]
Dear nailyk-fr

I am using this ROM:
https://forum.xda-developers.com/xperia-t-v/orig-development/rom-cyanogenmod-12-neo-s-homebuilds-t2984988

I must say that I have never heard about extract_elf_ramdisk and when I tried to learn about it I failed to discover understandable information. 
Would it be possible for you to help me out here? A year ago I was able to get the ROM I currently have thanks to guides but now I feel lost.

Thank you for your time.

**Comment by nailyk-fr** [comment]
I assume you wanna install this rom but have another one for now. 
The flashing process for most sony devices are flash boot.img (no need to flash twrp, you already have it) then use the extract_elf_ramdisk to boot into recovery: volume rocker while led bringup and until the color change. 

But better to ask in the xda thread I guess.

**Comment by nailyk-fr** [comment]
also why not flashing official 13.0 https://wiki.lineageos.org/devices/mint ?

**Comment by jakubpolo** [comment]
Actually right now I am using this ROM:
https://forum.xda-developers.com/xperia-t-v/orig-development/rom-cyanogenmod-12-neo-s-homebuilds-t2984988

And I want to move to this new one:
https://forum.xda-developers.com/xperia-t-v/orig-development/rom-aosp-nougat-xperia-t-tx-t3490463

I was told that I am in need of latest TWRP and that is why I am trying to get it.

**Comment by nailyk-fr** [comment]
Flash process looks like the same: 

> AOSP clean install :
> - (Optional) Flash the boot.img kernel from the ROM zip with Fastboot or Flashtool

**Comment by nailyk-fr** [comment]
Anyway really don't think it is a twrp issue.

**Comment by jakubpolo** [comment]
I did not try further steps because I thought that I need newer version of TWRP. So you think the 2.8.7.0 is enough?

**Comment by nailyk-fr** [comment]
yes. Use install steps into your xda threads to avoid troubles. 
This is not a twrp issue.

**Comment by jakubpolo** [comment]
OK, thanks.
